src/azureFunction/azureFunctionProjectService.ts

```diff
@@ -0,0 +1,165 @@
+				continue;
+			} else if (defaultLine.trimStart() === defaultBindingResult) { // Result change
+				replacedValueLines.push(defaultLine.replace(defaultBindingResult, sqlBindingResult));
+			} else {
+				// Normal lines to be included
+				replacedValueLines.push(defaultLine);
+			}
+		}
+		defaultBindedFunctionText = replacedValueLines.join(os.EOL);
+		fs.writeFileSync(filePath, defaultBindedFunctionText, 'utf-8');
+	}
+}
+
+// A C# Azure Functions project must be present in order to create a new Azure Function for the table
+async function isAzureFunctionProjectOpen(): Promise<boolean> {
+	if (vscode.workspace.workspaceFolders === undefined || vscode.workspace.workspaceFolders.length === 0) {
+		return false;
+	}
+	const projFile = await getProjectFile();
+	const hostFile = await getHostFile();
+	return projFile !== undefined && hostFile !== undefined;
+}
+
+// gets the azure functions project file path
+async function getProjectFile(): Promise<string | undefined> {
+	const projFiles = await vscode.workspace.findFiles('**/*.csproj');
+	return projFiles.length > 0 ? projFiles[0].fsPath : undefined;
+}
+
+// gets the host file config file path
+async function getHostFile(): Promise<string | undefined> {
+	const hostFiles = await vscode.workspace.findFiles('**/host.json');
+	return hostFiles.length > 0 ? hostFiles[0].fsPath : undefined;
+}
+
+// gets the local.settings.json file path
+async function getSettingsFile(): Promise<string | undefined> {
+	const settingsFiles = await vscode.workspace.findFiles('**/local.settings.json');
+	return settingsFiles.length > 0 ? settingsFiles[0].fsPath : undefined;
+}
+
+// retrieves the new function file once the file is created
+function getNewFunctionFile(): Promise<string> {
+	return new Promise((resolve) => {
+		const watcher = vscode.workspace.createFileSystemWatcher(new vscode.RelativePattern(
+			vscode.workspace.workspaceFolders[0], '**/*.cs'), false, true, true);
+		watcher.onDidCreate((e) => {
+			resolve(e.fsPath);
+			watcher.dispose();
+		});
+	});
+}
+
+// adds the required nuget package to the project
+async function addNugetReferenceToProjectFile(): Promise<void> {
+	// Make sure the nuget source is added
+	const currentSources = await executeCommand('dotnet nuget list source');
+	if (currentSources.indexOf(sqlBindingNugetSource) === -1) {
+		await executeCommand(`dotnet nuget add source ${sqlBindingNugetSource}`);
+	}
+	const projFile = await getProjectFile();
+	await executeCommand(`dotnet add package ${sqlBindingPackageName} --prerelease`, path.dirname(projFile));
+}
+
+// adds the Sql Connection String to the local.settings.json
+async function addConnectionStringToConfig(connectionString: string): Promise<void> {
+	const settingsFile = await getSettingsFile();
+	await azureFunctionUtils.setLocalAppSetting(path.dirname(settingsFile), constants.sqlConnectionString, connectionString);
+}
+
+async function executeCommand(command: string, cwd?: string): Promise<string> {
```

Comment:
This should go in a generic utils file ideally - it's not really specific to AF stuff